packages/cli/src/databases/dsl/Column.ts

```diff
@@ -0,0 +1,117 @@
+import type { Driver, TableColumnOptions } from 'typeorm';
+
+export class Column {
+	private type: 'int' | 'boolean' | 'varchar' | 'text' | 'json' | 'timestamp' | 'uuid';
+
+	private isGenerated = false;
+
+	private isNullable = true;
+
+	private isPrimary = false;
+
+	private length: number | 'auto';
+
+	private defaultValue: unknown;
+
+	constructor(private name: string) {}
+
+	get bool() {
+		this.type = 'boolean';
+		return this;
+	}
+
+	get int() {
+		this.type = 'int';
+		return this;
+	}
+
+	varchar(length?: number) {
+		this.type = 'varchar';
+		this.length = length ?? 'auto';
+		return this;
+	}
+
+	get text() {
+		this.type = 'text';
+		return this;
+	}
+
+	get json() {
+		this.type = 'json';
+		return this;
+	}
+
+	timestamp(length?: number) {
+		this.type = 'timestamp';
+		this.length = length ?? 'auto';
+		return this;
+	}
+
+	get uuid() {
+		this.type = 'uuid';
+		return this;
+	}
+
+	get primary() {
+		this.isPrimary = true;
+		return this;
+	}
+
+	get notNull() {
+		this.isNullable = false;
+		return this;
+	}
+
+	default(value: unknown) {
+		this.defaultValue = value;
+		return this;
+	}
+
+	get autoGenerate() {
+		this.isGenerated = true;
+		return this;
+	}
```

Comment:
Maybe we could remove the auto generate, unless it supports generating UUIDs. I think it's worth discouraging using auto increment integers WDYT?

Reply:
this was added to port over some of our older migrations. if we remove this, we can't convert those migrations to the DSL, and would have to wait until we decide to flatten the migrations at some point.

Reply:
Makes sense. Maybe we can add some sort of linting rule that would require an exception to prevent people from using auto increments going forward whenever possible.